src/components/Admin/Auth.js

```diff
@@ -0,0 +1,42 @@
+import React, {Component} from 'react';
+import initFirebase from '../../utils/firebase';
+import firebase from 'firebase/app';
+import 'firebase/auth';
+import 'firebase/firestore';
+
+class Auth extends Component {
+    state = {
+        status: 'Authenticating...'
+    }
+    
+    componentDidMount() {
+        initFirebase();
+        firebase.auth().getRedirectResult().then(result => {
+            if (result.user) {
+                const userRef = firebase.firestore().collection('users').doc(result.user.uid);
+                userRef.get().then(user => {
```

Comment:
Could we split this up so componentDidMount is just calling a bunch of init functions?

```
initFirebase();
authenticate(); // use a better name than this
```